CMS-Enterprise/mint-app
# [MINT-3157] Plan Data Exchange Data Loader

# MINT-3157

## Description

This PR introduces a dataloader for the new type, plan data exchange. It uses the new version of dataloaders with generics.
--> Note, there is now generic sort and error functionality, but it still remains in [PR](https://github.com/CMS-Enterprise/mint-app/pull/1402). Once that is merged, this loader should also implement that sorting functionality, and error handling instead of the manual approach that happens in this PR.
--> Note, the failing front end tests are not caused by this PR, the FE development is still in progress.

--> This PR also introduces a generic function to validate the new loader structure. Effectively, a test suite just has to ensure that the data exists, and then specify a set of data to compare per key, and a function to specify how to compare it. This will remove a lot of the boiler plate code needed to test a new data loader.

## How to test this change

1.  Seed the data base.
2.  Query model plans and retrieve the data exchange type.
3.  Ensure that the returned data matches the expected model plan
4. Ensure all unit tests pass.

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
